packages/uni_ui/lib/modal/widgets/enrollment_info.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:uni_ui/theme.dart';
+
+class ModalEnrollementInfo extends StatelessWidget {
+  const ModalEnrollementInfo({required this.enrollements});
+
+  final Map<String, String> enrollements;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        padding: const EdgeInsets.all(1.5),
+        decoration: BoxDecoration(
+            border: Border(
+                top: BorderSide(
+                    color: Theme.of(context).dividerColor, width: 1))),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text("Enrollments", style: Theme.of(context).textTheme.bodyMedium),
+            Wrap(
+              spacing: 1,
+              direction: Axis.horizontal,
+              children: createRows(context),
+            )
+          ],
+        ));
+  }
+
+  List<Widget> createRows(BuildContext context) {
+    return enrollements.entries.map((entry) {
```

Comment:
no, no, no. Do not create helper functions!!

Put the map directly in the top widget children, or create a new widget.
Suggestion: Listview.builder